### PR TITLE
JSON type infra

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [x] feature

## Modification
- thridparty: add velocypack and google benchmark
- define `JSON` PrimitiveType in thrift
- add JSON type in FE to make code compiled
- define several json functions

### CODE REVIEW

**Comment by mofeiatwork** [comment]
run starrocks_be_unittest

**Comment by mofeiatwork** [comment]
run starrocks_be_unittest

File: gensrc/script/vectorized/vectorized_functions.py [hunk 1]
```diff
@@ -472,14 +472,24 @@
...
+    # json type function
+    # TODO(mofei) change type to JSON
+    [110003, "parse_json", "VARCHAR", ["VARCHAR"], "JsonFunctions::parse_json"],
+    [110004, "json_string", "VARCHAR", ["VARCHAR"], "JsonFunctions::json_string"],
+    [110005, "json_query", "VARCHAR", ["VARCHAR", "VARCHAR"], "JsonFunctions::json_query"],
+    # [110006, "json_value", "JSON", ["JSON", "VARCHAR"], "JsonFunctions::json_query"],
+    [110007, "json_exists", "BOOLEAN", ["VARCHAR", "VARCHAR"], "JsonFunctions::json_exists"],
+    [110008, "json_object", "VARCHAR", ["VARCHAR", "..."], "JsonFunctions::json_object"],
+    [110009, "json_array", "VARCHAR", ["VARCHAR", "..."], "JsonFunctions::json_array"],
+
```
Comment (imay) [inline gensrc/script/vectorized/vectorized_functions.py:492]: I suggest you can comment these in this PR to avoid misusing.

Comment (mofeiatwork) [reply gensrc/script/vectorized/vectorized_functions.py:492]: This PR is just define these functions, to shared by FE and BE, the actual implementation will be submit later, with detailed comment.

Do you mean the function comment? Or a comment to warning others not using it ?

File: be/CMakeLists.txt [hunk 2]
```diff
@@ -55,11 +55,13 @@ set(BASE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
 set(ENV{STARROCKS_HOME} "${BASE_DIR}/../")
 set(BUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}")
 set(THIRDPARTY_DIR "$ENV{STARROCKS_THIRDPARTY}/installed/")
+set(STARROCKS_THIRDPARTY "$ENV{STARROCKS_THIRDPARTY}")
```
Comment (kangkaisen) [inline be/CMakeLists.txt:58]: Why need set this env variable ?

Comment (mofeiatwork) [reply be/CMakeLists.txt:58]: Clion IDE depends on it.

**Review by kangkaisen — APPROVED** [review]

File: be/CMakeLists.txt (hunk 2 above)
Comment (mofeiatwork) [reply be/CMakeLists.txt:58]: Specifically, in Clion remote development mode.

**Review by Seaven — APPROVED** [review]

File: gensrc/script/vectorized/vectorized_functions.py (hunk 1 above)
Comment (mofeiatwork) [reply gensrc/script/vectorized/vectorized_functions.py]: Finally, I decided to remove it, until a robust implementation.

**Pull request merged**